pkg/azure/provider/helpers/providerspec.go

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package helpers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+
+	"github.com/gardener/machine-controller-manager-provider-azure/pkg/azure/api"
+)
+
+// ExtractCloudConfiguration retrieves the name of the cloud instance to connect to from the AzureProviderSpec
+// and returns the corresponding azure cloud Configuration.
+func ExtractCloudConfiguration(spec *api.AzureProviderSpec) (cloud.Configuration, error) {
+	if spec.CloudConfiguration == nil {
+		return cloud.AzurePublic, nil
+	}
+
+	cloudConfigurationName := spec.CloudConfiguration.Name
+	switch {
+	case strings.EqualFold(cloudConfigurationName, "AzurePublic"):
+		return cloud.AzurePublic, nil
+	case strings.EqualFold(cloudConfigurationName, "AzureGovernment"):
```

Comment:
Also maybe its a good idea to have validation for the expected values, before we try to create the client